pkg/scheduler/framework/plugins/clusteraffinity/types_azure_test.go

```diff
@@ -0,0 +1,257 @@
+				Values:   []string{"5"},
+			},
+			wantHandled: false,
+		},
+		{
+			name:    "Azure SKU capacity property handled and available",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/capacity",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"2"},
+			},
+			wantHandled:   true,
+			wantAvailable: true,
+		},
+		{
+			name:    "Azure SKU capacity property handled but not available",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/NonExistentSKU/capacity",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"2"},
+			},
+			wantHandled:   true,
+			wantAvailable: false,
+		},
+		{
+			name:    "Azure SKU capacity property with invalid operator",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/capacity",
+				Operator: placementv1beta1.PropertySelectorEqualTo,
+				Values:   []string{"2"},
+			},
+			wantHandled: true,
+			wantErr:     true,
+		},
+		{
+			name:    "Azure SKU capacity property with non-integer value",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/capacity",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"two"},
+			},
+			wantHandled: true,
+			wantErr:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create mock server
+			server := createMockAttributeBasedVMSizeRecommenderServer(t, http.StatusOK)
+			defer server.Close()
+
+			client, err := compute.NewAttributeBasedVMSizeRecommenderClient(server.URL, http.DefaultClient)
+			if err != nil {
+				t.Fatalf("failed to create VM size recommender client: %v", err)
+			}
+
+			req := &clusterRequirement{
+				placementv1beta1.ClusterSelectorTerm{},
+				checker.NewPropertyChecker(*client),
+			}
+			handled, available, err := req.MatchPropertiesInPropertyChecker(tt.cluster, tt.selector)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MatchPropertiesInPropertyChecker() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if handled != tt.wantHandled {
+				t.Errorf("MatchPropertiesInPropertyChecker() handled = %v, want %v", handled, tt.wantHandled)
+			}
+			if handled {
+				if available != tt.wantAvailable {
+					t.Errorf("MatchPropertiesInPropertyChecker() available = %v, want %v", available, tt.wantAvailable)
+				}
+			}
+		})
+	}
+}
+
+// createMockAttributeBasedVMSizeRecommenderServer creates a mock HTTP server for testing AttributeBasedVMSizeRecommenderClient.
+func createMockAttributeBasedVMSizeRecommenderServer(t *testing.T, httpStatusCode int) *httptest.Server {
```

Comment:
duplicate code of property checker test code?

if so, please refactor it and move to test util packages.